src/hotspot/share/runtime/disableStackTracingMark.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_RUNTIME_DISABLESTACKTRACINGMARK_HPP
+#define SHARE_RUNTIME_DISABLESTACKTRACINGMARK_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/macros.hpp"
+
+class JavaThread;
+
+// Use this class to mark a section of code where stack tracing is not safe
+// and should be avoided.
+class DisableStackTracingMark : public StackObj {
```

Comment:
One thing that could perhaps be a bit of a gotcha with this is the name selection, "DisableStackTracingMark". One could get the impression that this disables stack traces entirely. But this is not the case for asynchronous operations, like AsyncGetCallTrace and JFR - they fetch the information from the CPU context to produce a stacktrace, if there is no ljf. Not a biggie, but might be confusing. Perhaps "NoLastJavaFrame" or something?

Reply:
Good point. I'll try to think of a better name.

Reply:
Trying out ClearFrameAnchorMark in the latest commit - what do you think?

Reply:
I am always slightly jarred about the discrepancy between the API, set_last_java_frame() and reset_last_java_frame() and the fact that the struct field is called _frame_anchor. Conceptually, the "last java frame" is better. Therefore, I would suggest ClearLastJavaFrameMark. Sorry.

Reply:
Okay, let's take that for a spin.